src/servers/src/mysql/handler.rs

```diff
@@ -105,14 +116,47 @@ impl MysqlInstanceShim {
         output
     }
 
-    fn set_query(&self, query: String) -> u32 {
-        let stmt_id = self.prepared_stmts_counter.fetch_add(1, Ordering::SeqCst);
-        let mut guard = self.prepared_stmts.write();
-        guard.insert(stmt_id, query);
+    /// Execute the logical plan and return the output
+    async fn do_exec_plan(&self, query: &str, plan: LogicalPlan) -> Vec<Result<Output>> {
+        trace!("Start executing query: '{}'", query);
+        let start = Instant::now();
+
+        let output =
+            if let Some(output) = crate::mysql::federated::check(query, self.session.context()) {
+                vec![Ok(output)]
+            } else {
+                vec![
+                    self.query_handler
+                        .do_exec_plan(plan, self.session.context())
+                        .await,
+                ]
+            };
+
+        trace!(
+            "Finished executing query: '{}', total time costs in microseconds: {}",
+            query,
+            start.elapsed().as_micros()
+        );
```

Comment:
Maybe a `timer!()` is more suitable?

Reply:
It's copied form #1162  and looks unnecessary, I'll remove it.
Already have a timer in SQL handler.